### PR TITLE
[T3A2 - T3A4][W13-A2] Albert Yeoh

### PR DESCRIPTION
**Pull request by bertojo**
Originally from tutorial 3.
Ready for review.

### CODE REVIEW

**Comment by bertojo** [comment]
Sorry, i forgot to branch out before editing the code.

**Comment by okkhoy** [comment]
good, you realized your mistake. going forward avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -25,7 +32,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        // a/BLOCK, STREET, UNIT, POSTAL_CODE
+        String[] components = address.split(",");
+        Block block = new Block(components[BLOCK].substring(2, components[BLOCK].length()).trim());
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: all your constants are initialized to zero above; how does the indexing work properly here?

File: src/seedu/addressbook/data/person/Block.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package seedu.addressbook.data.person;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: Please include a header comment to indicate what this class/method does?

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 import seedu.addressbook.data.exception.IllegalValueException;
 
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: you need to update:
1. user guide
2. help within the program
3. test files

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -25,7 +32,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        // a/BLOCK, STREET, UNIT, POSTAL_CODE
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: don't you have to do some different form of validation after this update?

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -25,7 +32,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+        String[] components = address.split(",");
+        Block block = new Block(components[BLOCK].substring(2, components[BLOCK].length()).trim());
+        Street street = new Street(components[STREET].trim());
+        Unit unit = new Unit(components[UNIT].trim());
+        PostalCode postal = new PostalCode(components[POSTAL].trim());
         this.value = address;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: why is `Address.value` still retained? you will only introduce duplicity and inconsistency by retaining this variable.

**Comment by okkhoy** [comment]
@bertojo 
Some comments added. Please ack & close the PR after reading comments.

**Pull request closed without merging**